demos/custom-tabs-headers-demo/src/main/java/org/chromium/customtabsdemos/MainActivity.java

```diff
@@ -0,0 +1,71 @@
+package org.chromium.customtabsdemos;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.browser.customtabs.CustomTabsCallback;
+import androidx.browser.customtabs.CustomTabsClient;
+import androidx.browser.customtabs.CustomTabsIntent;
+import androidx.browser.customtabs.CustomTabsService;
+import androidx.browser.customtabs.CustomTabsServiceConnection;
+import androidx.browser.customtabs.CustomTabsSession;
+import android.content.ComponentName;
+import android.net.Uri;
+import android.os.Bundle;
+import android.provider.Browser;
+
+public class MainActivity extends AppCompatActivity {
+
+    private CustomTabsSession mSession;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        String url = "https://padr31.github.io/";
+
+        findViewById(R.id.btnExtra).setOnClickListener(e -> {
+
+            // Set up a callback that launches the cross origin intent after session validated.
+            CustomTabsCallback callback = new CustomTabsCallback() {
+                @Override
+                public void onRelationshipValidationResult(int relation, @NonNull Uri requestedOrigin, boolean result, @Nullable Bundle extras) {
+                    super.onRelationshipValidationResult(relation, requestedOrigin, result, extras);
+                    // Launch custom tabs intent after session was validated as the same origin.
+                    CustomTabsIntent intent = constructExtraHeadersIntent(mSession);
+                    intent.launchUrl(MainActivity.this, Uri.parse(url));
+                }
+            };
+
+            // Set up a connection that warms up and validates a session.
+            CustomTabsServiceConnection connection = new CustomTabsServiceConnection() {
+                @Override
+                public void onCustomTabsServiceConnected(@NonNull ComponentName name, @NonNull CustomTabsClient client) {
```

Comment:
One thing to consider is happens if the user:
- Clicks the button and opens the link
- Closes the  Custom Tab
- Clicks the button to open the link again.

It seems right now:
-  We're not disconnecting from the Custom Tabs service and seems like we'd be calling `bindCustomTabsService` again.
- We probably don't want to disconnect / reconnect to the Custom Tabs service, but use the same connection with a different session, if the link is different, for instance. I'm not sure if this is possible though.